applications/StructuralMechanicsApplication/custom_elements/spring_damper_element.cpp

```diff
@@ -0,0 +1,566 @@
+//************************************VELOCITY****************************************
+//************************************************************************************
+
+template<std::size_t TDim>
+void SpringDamperElement<TDim>::GetFirstDerivativesVector( Vector& rValues, int Step ) const
+{
+    //GIVES THE VECTOR WITH THE TIME DERIVATIVE OF THE DOFS VARIABLES OF THE ELEMENT (i.e. ELEMENT VELOCITIES)
+    if ( rValues.size() != msElementSize )
+    {
+        rValues.resize( msElementSize, false );
+    }
+
+    for ( std::size_t i = 0; i < GetGeometry().size(); ++i)
+    {
+        const array_1d<double, 3>& vel = GetGeometry()[i].FastGetSolutionStepValue( VELOCITY, Step );
+        const array_1d<double, 3>& avel = GetGeometry()[i].FastGetSolutionStepValue( ANGULAR_VELOCITY, Step );
+        std::size_t index = i * msLocalSize;
+
+        if constexpr (TDim == 2) {
+            rValues[index] = vel[0];
+            rValues[index + 1] = vel[1];
+            rValues[index + 2] = avel[2];
+        }
+        else if constexpr (TDim == 3) {
+            rValues[index] = vel[0];
+            rValues[index + 1] = vel[1];
+            rValues[index + 2] = vel[2];
+            rValues[index + 3] = avel[0];
+            rValues[index + 4] = avel[1];
+            rValues[index + 5] = avel[2];
+        }
+
+    }
+}
+
+//*********************************ACCELERATION***************************************
+//************************************************************************************
+
+template<std::size_t TDim>
+void SpringDamperElement<TDim>::GetSecondDerivativesVector( Vector& rValues, int Step ) const
+{
+    //GIVES THE VECTOR WITH THE TIME SECOND DERIVATIVE OF THE DOFS VARIABLES OF THE ELEMENT (i.e. ELEMENT ACCELERATIONS)
+    if ( rValues.size() != msElementSize )
+    {
+        rValues.resize( msElementSize, false );
+    }
+
+    for ( std::size_t i = 0; i < GetGeometry().size(); ++i)
+    {
+        const array_1d<double, 3>& acc = GetGeometry()[i].FastGetSolutionStepValue( ACCELERATION, Step );
+        const array_1d<double, 3>& aacc = GetGeometry()[i].FastGetSolutionStepValue( ANGULAR_ACCELERATION, Step );
+        std::size_t index = i * msLocalSize;
+
+        if constexpr (TDim == 2) {
+            rValues[index] = acc[0];
+            rValues[index + 1] = acc[1];
+            rValues[index + 2] = aacc[2];
+        }
+        else if constexpr (TDim == 3) {
+            rValues[index] = acc[0];
+            rValues[index + 1] = acc[1];
+            rValues[index + 2] = acc[2];
+            rValues[index + 3] = aacc[0];
+            rValues[index + 4] = aacc[1];
+            rValues[index + 5] = aacc[2];
+        }
+    }
+}
+
+
+//************* COMPUTING  METHODS
+//************************************************************************************
+//************************************************************************************
+
+template<std::size_t TDim>
+void SpringDamperElement<TDim>::CalculateLocalSystem(
```

Comment:
hm somehow the comment didnt show up, so here it is again:

Please take the chance to add `const` versions of these functions, like it is done in [the beam](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp#L1005-L1037)

this helps to avoid bugs related to non-const access e.g. of Properties and DataValueContainer

I.e. add `ConstCalculateLocalSystem(...) const`, which gets called from `CalculateLocalSystem`